src/JSONPrinting.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+#include "KafkaMessageMetadataStruct.h"
+#include <algorithm>
+#include <nlohmann/json.hpp>
```

Comment:
`algorithm` and `json` headers are only needed in the `.cpp`, try to keep header files as clean as possible.